.github/workflows/ci.yml

```
@@ -191,88 +187,90 @@ jobs:
           ./zcutil/build.sh
           -j"${{ steps.nproc.outputs.count }}"
 
-      - name: Authenticate to Google Cloud
-        uses: google-github-actions/auth@v2
-        with:
-          credentials_json: ${{ secrets.GCP_SA_KEY }}
-
       - name: Upload zcashd artifact
-        uses: google-github-actions/upload-cloud-storage@v2
+        uses: actions/upload-artifact@v4
         with:
+          name: ${{ format('./src/zcashd{0}', matrix.file_ext) }}-${{ matrix.name }}
           path: ${{ format('./src/zcashd{0}', matrix.file_ext) }}
```

```suggestion
          path: |
              ${{ format('./src/zcashd{0}', matrix.file_ext) }}
              ${{ format('./src/zcash-cli{0}', matrix.file_ext) }}
              ${{ format('./src/zcashd-wallet-tool{0}', matrix.file_ext) }}
              ${{ format('./src/zcash-inspect{0}', matrix.file_ext) }}
              ${{ format('./src/zcash-tx{0}', matrix.file_ext) }}
              ${{ format('./src/test/test_bitcoin{0}', matrix.file_ext) }}
              ${{ format('./src/zcash-gtest{0}', matrix.file_ext) }}
              ./src/test/buildenv.py-${{ matrix.name }}
```
The `src/` will be stripped because it's common to all of these paths, but that's ok.